WIP - Security update (#1111)
shopware/docs: resources/references/security.md

This reference presents a comprehensive compilation of all security measures implemented in Shopware 6, along with instructions on how to configure them.

## ACL in the Administration

> The Access Control List (ACL) in Shopware ensures that by default, data can only be created, read, updated, or deleted ( CRUD), once the user has specific privileges for a module. [ACL in the Administration](../concepts/framework/architecture/administration-concept.md#acl-in-the-administration)

Reviewer: ```suggestion
The Access Control Layer (ACL) in Shopware ensures that by default, data can only be created, read, updated, or deleted (CRUD), once the user has specific privileges for a module. [ACL in the Administration](../concepts/framework/architecture/administration-concept.md#acl-in-the-administration)
```

Author: As discussed in slack we leave it with "List" - I will change the space as a new commit and resolve the conversation.

## API aware fields

The `ApiAware` flag allows you to control what fields of your entity are exposed to the Store API. For more information, refer to [Flags Reference](core-reference/dal-reference/flags-reference.md).

## Captchas

Captchas helps verify the user's humanity and prevents automated bots or scripts from gaining access. For more information, refer to [Captchas](https://docs.shopware.com/en/shopware-en/settings/basic-information#captcha)

> ## CORS

Reviewer 2: ```suggestion
## CSP

[Content Security Policies](https://developer.mozilla.org/en-US/docs/Web/HTTP/CSP) are used to prevent Cross-Site-Scripting (XSS) attacks, as well as data injection attacks. The content security policy specify from which sources additional content (e.g. images, scripts, ...) can be included.

The default policies are configured over the `shopware.security.csp_templates` symfony container parameter, and can be adjusted over the container configuration.
```

@shyim did i miss something?

This reference presents a comprehensive compilation of all security measures implemented in Shopware 6, along with instructions on how to configure them.

## ACL in the Administration

The Access Control List (ACL) in Shopware ensures that by default, data can only be created, read, updated, or deleted (CRUD), once the user has specific privileges for a module. [ACL in the Administration](../../concepts/framework/architecture/administration-concept.md#acl-in-the-administration)

> ## API aware fields

Reviewer 3: Please mention everything either in singular or plural. IMO, singular would be good like "API aware field"

The `ApiAware` flag allows you to control what fields of your entity are exposed to the Store API. For more information, refer to [Flags Reference](core-reference/dal-reference/flags-reference.md).

## Captchas

> Captchas helps verify the user's humanity and prevents automated bots or scripts from gaining access. For more information, refer to [Captchas](https://docs.shopware.com/en/shopware-en/settings/basic-information#captcha)

Reviewer 3: Singular/plural

## CSP

> [Content Security Policies](https://developer.mozilla.org/en-US/docs/Web/HTTP/CSP) are used to prevent Cross-Site-Scripting (XSS) attacks, as well as data injection attacks. The content security policy specify from which sources additional content (e.g. images, scripts, ...) can be included.

Reviewer 3: ```suggestion
[Content Security Policies](https://developer.mozilla.org/en-US/docs/Web/HTTP/CSP) (CSPs) are used to prevent Cross-Site-Scripting (XSS) attacks, as well as data injection attacks. This policy specifies the sources from which additional content (e.g., images, scripts, etc.) can be included.
```

> The default policies are configured over the `shopware.security.csp_templates` symfony container parameter, and can be adjusted over the container configuration.

Reviewer 3: ```suggestion
The default policies are configured over the `shopware.security.csp_templates` symfony container parameter and can be adjusted over the container configuration.
```

> TBD

Reviewer 3: Why TBD mentioned here?

## File access

Shopware 6 stores and processes a wide variety of files. This goes from product images or videos to generated documents such as invoices or delivery notes. This data should be stored securely, and backups should be generated regularly. For more information, refer to [File system](/docs/guides/hosting/infrastructure/filesystem)

## GDPR compliance

> GDPR (General Data Protection Regulation) is a comprehensive European Union (EU) regulation that enhances individuals' privacy rights by imposing strict rules on how organizations collect, process, and protect personal data. For more information, refer to [GDPR](https://docs.shopware.com/en/shopware-6-en/tutorials-and-faq/gdpr) guide.

Reviewer 3: ```suggestion
General Data Protection Regulation (GDPR) is a comprehensive European Union (EU) regulation that enhances individuals' privacy rights by imposing strict rules on how organizations collect, process, and protect personal data. For more information, refer to [GDPR](https://docs.shopware.com/en/shopware-6-en/tutorials-and-faq/gdpr) guide.
```

## Reset sessions when changing password

As soon as a password is changed for a user or customer the session is invalid and the user or customer must log in again. For more information, refer to:

- [User Changelog](https://github.com/shopware/platform/commit/5ea99ee5d7a12bab3a01a64c3948eee7c4188ede)
- [Customer Changelog](https://github.com/shopware/platform/commit/47b4b094c13f62db860be2f431138bb45c0bd0b6)

## SameSite cookies

SameSite prevents the browser from sending cookies along with cross-site requests. For more information on this, refer to [SameSite Protection](../../../guides/hosting/configurations/framework/samesite-protection.md).

## Security plugin

Obtaining security fixes without version upgrades is possible through the [Security plugin](../../../guides/hosting/installation-updates/cluster-setup.md#security-plugin).

## Storefront IP Whitelisting

To enable access even during maintenance mode IP addresses can be added [Storefront IP whitelisting](https://docs.shopware.com/en/shopware-6-en/settings/saleschannel#status).

## SQL injection

> SQL injection allows an attacker to execute new or modify existing SQL statements to access information that they are not allowed to access. [SQL injection](https://www.doctrine-project.org/projects/doctrine-dbal/en/current/reference/security.html)

Reviewer 2: ```suggestion
SQL injection allows an attacker to execute new or modify existing SQL statements to access information that they are not allowed to access. By mainly using our own [Data Abstraction Layer](https://developer.shopware.com/docs/concepts/framework/data-abstraction-layer.html), that does not expose SQL directly, most of the SQL injection attack vectors are prevented. Whenever direct SQL is being used, the [best practices from Doctrine DBAL](https://www.doctrine-project.org/projects/doctrine-dbal/en/current/reference/security.html) are followed to ensure proper escaping of user input.
```

Shopware 6 provides certain rate limits by default that reduces the risk of brute-force attacks for pages like login or password reset. For more information, refer to [Rate Limiter](../../../guides/hosting/infrastructure/rate-limiter.md) guide.

## Reset sessions when changing password

As soon as a password is changed for a user or customer the session is invalid and the user or customer must log in again. For more information, refer to:

- [User Changelog](https://github.com/shopware/platform/commit/5ea99ee5d7a12bab3a01a64c3948eee7c4188ede)
- [Customer Changelog](https://github.com/shopware/platform/commit/47b4b094c13f62db860be2f431138bb45c0bd0b6)

## SameSite cookies

SameSite prevents the browser from sending cookies along with cross-site requests. For more information on this, refer to [SameSite Protection](../../../guides/hosting/configurations/framework/samesite-protection.md).

## Security plugin

Obtaining security fixes without version upgrades is possible through the [Security plugin](../../../guides/hosting/installation-updates/cluster-setup.md#security-plugin).

## Storefront IP Whitelisting

> To enable access even during maintenance mode IP addresses can be added [Storefront IP whitelisting](https://docs.shopware.com/en/shopware-6-en/settings/saleschannel#status).

Reviewer 3: ```suggestion
To enable access even during maintenance mode, IP addresses can be added to [Storefront IP whitelisting](https://docs.shopware.com/en/shopware-6-en/settings/saleschannel#status).
```
